tensorboard/data/provider.py

```diff
@@ -486,6 +516,78 @@ def __repr__(self):
         )
 
 
+class HyperparameterDomainType(enum.Enum):
+    """Describes how to represent the set of known values for a hyperparameter."""
+
+    # A range of numeric values. Normally represented as Tuple[float, float].
+    INTERVAL = "interval"
+    # A finite set of numeric values. Normally represented as List[float].
+    DISCRETE_FLOAT = "discrete_float"
+    # A finite set of string values. Normally represented as List[string].
+    DISCRETE_STRING = "discrete_string"
+    # A finite set of bool values. Normally represented as List[bool].
+    DISCRETE_BOOL = "discrete_bool"
+
+
+class Hyperparameter:
+    """Metadata about a hyperparameter.
+
+    Attributes:
+      hyperparameter_name: A string identifier for the hyperparameter that should be unique
```

Comment:
Since this is under the `Hyperparameter` class, maybe we can omit the `hyperparameter` prefix to be more concise? Ditto the display name below.

Reply:
The choice of name here is to be consistent with other classes in this file. `Run` has `run_id` and `run_name`. `ExperimentMetadata` has `experiment_name` and `experiment_description`.

Reply:
Ack.